test/test_modules.py

```diff
@@ -108,6 +111,68 @@ def test_factory_kwargs(self, device, dtype, module_info):
                             buffer.dtype, dtype,
                             f'Buffer {name} is of dtype {buffer.dtype} instead of the expected dtype {dtype}')
 
+    @modules(module_db)
+    def test_check_inplace(self, device, dtype, module_info):
+        # Check if the inplace variant of the module gives the same result as the out of place.
+        module_cls = module_info.module_cls
+        if 'inplace' not in signature(module_cls).parameters:
+            return
+
+        # check_inplace doesn't support multiple input tensors, since we don't have any modules
+        # that modify the inputs in-place and that accept more than one input
+        module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
+                                                       requires_grad=True)
+        for module_input in module_inputs:
+            if module_input.forward_input is None:
+                continue
+
+            input_args, input_kwargs = module_input.forward_input.args, module_input.forward_input.kwargs
+
+            # there aren't any modules that that modify that inputs in-place and accepts multiple
+            # inputs
+            if len(input_args) != 1:
+                continue
+            input = input_args[0]
+
+            # === Instantiate the module. ===
+            args, kwargs = module_input.constructor_input.args, module_input.constructor_input.kwargs
+            kwargs = deepcopy(kwargs)
+            kwargs["inplace"] = False
+            m = module_cls(*args, **kwargs)
+            m.to(device).to(dtype)
+
+            kwargs_ip = deepcopy(kwargs)
+            kwargs_ip["inplace"] = True
+            m_ip = module_cls(*args, **kwargs_ip)
+            m_ip.to(device).to(dtype)
+
+            # === Check that the module performs a inplace operation ===
```

Comment:
I don't think this comment describes what's going on here, since it looks like this section does the out-of-place forward